File: scratch/sync_csv.py
import csv
import json
import os
import uuid
from datetime import datetime

csv_path = 'sample_tasks.csv'
json_path = 'data/tasks.json'

tasks = []
if os.path.exists(csv_path):
    with open(csv_path, mode='r', encoding='utf-8') as f:
        reader = csv.DictReader(f)
        for row in reader:
            task = {
                "id": str(uuid.uuid4()),
                "createdAt": datetime.now().isoformat(),
                "completed": False,
                "userId": "owner",
                "title": row.get('Title', 'Untitled'),
                "room": row.get('Room', 'General'),
                "category": row.get('Category', 'other').lower(),
                "priority": row.get('Priority', 'medium').lower(),
                "price": float(row.get('Price', 0)),
                "difficulty": int(row.get('Difficulty', 1)),
                "time": float(row.get('Time', 1)),
                "repeat": row.get('Repeat', 'false').lower() == 'true'
            }
            tasks.append(task)

with open(json_path, 'w', encoding='utf-8') as f:
    json.dump(tasks, f, indent=2)

print(f"Imported {len(tasks)} tasks from CSV to JSON.")

File: scratch/sync_csv.js
const fs = require('fs');
const path = require('path');

const csvPath = path.join(__dirname, 'sample_tasks.csv');
const tasksJsonPath = path.join(__dirname, 'data', 'tasks.json');

const csvData = fs.readFileSync(csvPath, 'utf8');
const lines = csvData.trim().split('\n');
const headers = lines[0].split(',');

const tasks = lines.slice(1).map(line => {
    const values = line.split(',');
    const task = {
        id: Math.random().toString(36).substr(2, 9), // Simple ID generator for script
        createdAt: new Date().toISOString(),
        completed: false,
        userId: 'admin' // Placeholder
    };
    
    headers.forEach((header, index) => {
        let val = values[index];
        const key = header.toLowerCase();
        
        if (key === 'price' || key === 'difficulty' || key === 'time') {
            val = parseFloat(val) || 0;
        } else if (key === 'repeat') {
            val = val === 'true';
        }
        
        task[key] = val;
    });
    
    return task;
});

fs.writeFileSync(tasksJsonPath, JSON.stringify(tasks, null, 2));
console.log(`Imported ${tasks.length} tasks from CSV to JSON.`);
